addons/website_event/views/snippets/s_events.xml

```diff
@@ -0,0 +1,107 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+
+<template id="s_events" name="Events">
+    <t t-call="website.s_dynamic_snippet_template">
+        <t t-set="snippet_name" t-value="'s_events'"/>
+        <t t-set="snippet_classes" t-value="'s_event_upcoming_snippet'"/>
+    </t>
+</template>
+
+<!--TEMPLATES-->
+
+<template id="dynamic_filter_template_event_event_picture" name="Picture Layout" priority="5">
+    <figure t-foreach="records" t-as="data" class="w-100 s_events_event">
+        <t t-set="record" t-value="data['_record']"/>
```

Comment:
Can't we set record as ``data['_record']._set_tz_context()`` ? That way we don't have to repeat it :)